configure.ac

```diff
@@ -41,6 +41,15 @@ The executable may also be named 'apxs'.
 ])
 fi
 
+AC_ARG_WITH(
```

Comment:
Agreed, changed to use enable instead. But do you also want the macro changed from WITH_DIAGNOSTICS to ENABLE_DIAGNOSTICS? It seems like for consistency it should be the later.

But this brings up another question for me. When I was first adding this to configure.ac it seemed like the way the defines were handled by appending -Dxxx to xxx_CFLAGS was non-standard and produced really long compile arguments. Shouldn't we be using AC_DEFINE with autoheader and include config.h in auth_mellon.h? That seems closer to what is considered best practice. I only did it the other way to remain consistent with what was already being done in configure.ac

Reply:
I don't have strong feelings either way wrt. the macro name. If you can easily change it, I do like consistency, but it isn't really important; I think it is unlikely to confuse anyone.

Wrt. switching to autoheader & `config.h`, I am not against switching to the standard way of doing things. The reason it is the way it is at the moment is simply because I am not very familiar with autoconf and the associated tools.

Reply:
I changed all the WITH_DIAGNOSTICS to ENABLE_DIAGNOSTICS, it was an easy modification (gotta love git).

I'll do the patch for switching to autoheader separately, I already have a working version, but best not to mix an unrelated change into these commits.